thrust/thrust/iterator/counting_iterator.h

```diff
@@ -186,6 +186,19 @@ struct counting_iterator_equal<Difference,
   }
 };
 
+struct empty
+{};
+
+template <typename T>
+struct value_holder
+{
+  T value;
+
+  _CCCL_HOST_DEVICE auto operator()() const
+  {
+    return value;
+  }
+};
```

Comment:
I believe we should be able to avoid this by using the mdspan helper `__de_ice` available in `<cuda\std\__mdspan\submdspan_helper.h>`

Reply:
This is exactly why I have put this draft PR up. I was certain, we would already have a helper somewhere. Thx!